### DISCUSSION TITLE
Parallelize Multiple Subgraph Matching

### DISCUSSION THREAD

**Discussion by skmda37** [pyg-team/pytorch_geometric#7130] [Q&A]
Suppose I have two batches of graphs $(G_1,...,G_n)$ and  $(S_1,...,S_n)$. I want to check whether $S_i$ is a subgraph of $G_i$ for each $i=1,...,n$. Is there a method that parallelizes this over the batch? My naive attempt would have been a for loop and using VF2 from Networkx (https://networkx.org/documentation/stable/reference/algorithms/isomorphism.vf2.html) to check for the subgraph in each iteration.

Eventually I would also really like to be able to extract multiple subgraph matchings. Is someone by any chance also familiar with a method that allows to check and extract multiple subgraph matchings? As far as I understand, VF2 in networkx only returns one matching. How can I get a second potential matching? Also as far as I understand pytorch_geometric does not have an implementation of subgraph matching, correct?

Thanks in advance!

**Comment by rusty1s** [comment]
Yeah, we don't have a method for exact subgraph matching since PyG is a GNN/deep learning library, not a graph algorithm library. For now, you have to rely on `networkx` or similar libraries for this job, sorry :(
